python_builtin_functions.py

```diff
@@ -5,84 +5,167 @@ def myabs(number):
-    pass
+    minimum=iterable[0]
+    for item in range(1,len(iterable)):
+        if iterable[item]<minimum:
+            minimum=iterable[item]
+    return minimum
 
 
-def mydivmod(a, b):
+def mydivmod(num1, num2):
     """ Implementation of built-in `divmod` function """
-    pass
+    return (num1//num2,num1%num2)
 
 
 def mybool(x):
     """ Implementation of built-in `bool` function """
-    pass
+    return True if x else False
 
 
 def mypow(base, exp):
     """ Implementation of built-in `pow` function """
-    pass
+    counter=1
+    prod=1
+    while counter<=exp:
+        prod=prod*base
+        counter+=1
+    return prod
 
 
 def myreversed(seq):
     """ Implementation of built-in `reversed` function """
-    pass
+    reverse=seq[::-1]
+    return reverse
 
 
 def myenumerate(iterable, start=0):
     """ Implementation of built-in `enumerate` function """
-    pass
+    lst=[]
+    for i in range(len(iterable)):
+        lst.append((i,iterable[i]))
+    return(lst)
 
 
 def mybin(x):
     """ Implementation of built-in `bin` function """
-    pass
+    lst=['0b']
+    while(x>0):
+        r=x%2
+        x=x//2
+        lst.insert(1,str(r))
+    return "".join(lst)
 
 
 def myhex(x):
     """ Implementation of built-in `hex` function """
-    pass
-
+    lst=['0x']
+    while(x>0):
+        r=x%16
+        x=x//16
+        if r==10:
+            r='a'
+        elif r==11:
+            r='b'
+        elif r==12:
+            r='c'
+        elif r==13:
+            r='d'
+        elif r==14:
+            r='e'
+        elif r==15:
+            r='f'
+        else:
+            r=r
+        lst.insert(1,str(r))
+    return ''.join(lst)
 
 def myoct(x):
     """ Implementation of built-in `oct` function """
-    pass
-
+    lst=['0o']
+    while(x>0):
+        r=x%8
+        x=x//8
+        lst.insert(1,str(r))
+    return "".join(lst)
 
 def myisinstance(object, classinfo):
     """ Implementation of built-in `isinstance` function """
-    pass
+    t1=type(object)
+    t2=str(t1).find(classinfo)
+    if t2>=0:
+        return True
+    return False
 
 
 def myround(number, ndigits=None):
     """ Implementation of built-in `round` function """
-    pass
+    dot='.'
+    number=str(number)
+    lst=list(str(number))
+    decimal=str(number).find(dot)
+    if decimal<0:
+        return number
+    else:
+        if int(len(number[(decimal+1):]))<=ndigits:
+            return number
+        else:
+            lastdigit=decimal+ndigits+1
+            if int(lst[lastdigit])>5:
+                lst[lastdigit-1]=str(int(lst[lastdigit-1])+1)
+            return ''.join(lst[0:lastdigit])
 
 
 def mysorted(iterable, reverse=False):
     """ Implementation of built-in `sorted` function """
-    pass
\ No newline at end of file
+    iterable=list(iterable)
+    #print(len(lst))
+    for i in range(len(iterable)):
+        for j in range(len(iterable)):
+            if i!=j:
+                if iterable[i]>iterable[j]:
```

Comment:
in Python, for swapping you can use `tuple packing - tuple unpacking` concept.

EG: If you want to swap `a` and `b`, then you can do so by just
`a, b = b, a`. No need of temporary variable.